src/components/HoldPages/HoldRequestErrorBanner.tsx

```diff
@@ -0,0 +1,177 @@
+ */
+const HoldRequestErrorBanner = ({
+  item,
+  errorStatus = "failed",
+  patronEligibilityStatus,
+}: HoldRequestErrorBannerProps) => {
+  const { onOpen, setItemMetadata } = useContext(FeedbackContext)
+
+  const onContact = (metadata: ItemMetadata) => {
+    setItemMetadata(metadata)
+    onOpen()
+  }
+
+  return (
+    <Banner
+      type="negative"
+      heading={HOLD_PAGE_ERROR_HEADINGS?.[errorStatus] || null}
+      data-testid="hold-request-error"
+      // TODO: Ask DS team to make button link variant match the default link styles
+      sx={{
+        button: {
+          color: "ui.error.primary",
+          "&:hover": { color: "ui.error.primary" },
+        },
+      }}
+      content={
+        <>
+          <Box>
+            {HOLD_PAGE_CONTACT_PREFIXES?.[errorStatus] ? (
+              <Text>
+                <>
+                  {HOLD_PAGE_CONTACT_PREFIXES?.[errorStatus]}
+                  {" Please try again, "}
+                  <Button
+                    id="hold-contact"
+                    onClick={() =>
+                      onContact({
+                        id: item.id,
+                        barcode: item.barcode,
+                        callNumber: item.callNumber,
+                        bibId: item.bibId,
+                        notificationText: `Request failed for call number ${item.callNumber}`,
+                      })
+                    }
+                    buttonType="link"
+                    // TODO: Ask DS team to make button link variant match the default link styles
+                    sx={{
+                      display: "inline",
+                      fontWeight: "inherit",
+                      fontSize: "inherit",
+                      p: 0,
+                      height: "auto",
+                      textAlign: "left",
+                      minHeight: "auto",
+                      textDecorationStyle: "dotted",
+                      textDecorationThickness: "1px",
+                      textUnderlineOffset: "2px",
+                    }}
+                  >
+                    contact us
+                  </Button>{" "}
+                  for assistance, or{" "}
+                  <RCLink href="/search">start a new search.</RCLink>
+                </>
+              </Text>
+            ) : null}
+            {(() => {
+              switch (errorStatus) {
+                case "invalid":
+                  return "Some fields contain errors. Please correct and submit again."
+                case "patronIneligible":
+                  return patronEligibilityStatus ? (
+                    <PatronErrors
+                      patronEligibilityStatus={patronEligibilityStatus}
+                    />
+                  ) : null
+                default:
+                  return null
+              }
+            })()}
+          </Box>
+        </>
+      }
+      mb="s"
+    />
+  )
+}
+
+const PatronErrors = ({
```

Comment:
This component might get lost within the file and maybe it can be its own file. Even though it's a small component, I think you can break out the logic from the component.